src/pages/index.js

```diff
@@ -4,73 +4,220 @@ import Section from "../components/Section.js";
+      userInformation.setUserInfo(userInfo);
+      userInformation.setUserAvatar(userInfo);
+      if (Array.isArray(initialCards)) {
+        cardSection = new Section({ renderer: renderCard }, ".cards__list");
+        cardSection.renderItems(initialCards);
+      } else {
+        console.error("Initial cards data is not an array:", initialCards);
+      }
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 
-constants.addCardButton.addEventListener("click", () => {
-  addCardForm.open();
-});
+  function renderCard(cardData) {
+    const card = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    ).getView();
+    cardSection.addItem(card);
+  }
 
-const cardSection = new Section(
-  {
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardSection.addItem(cardElement);
-    },
-  },
-  constants.cardListEl
-);
+  const cardPreview = new PopupWithImage("#image-popup");
+  const confirmDeletePopup = new PopupWithConfirm("#delete-card-popup");
+  cardPreview.setEventListeners();
+
+  const profileEditValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#profile-form")
+  );
+  const addCardValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#add-card-form")
+  );
+  const avatarValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#avatar-form")
+  );
+
+  profileEditValidator.enableValidation();
+  addCardValidator.enableValidation();
+  avatarValidator.enableValidation();
+
+  const avatarImagePopup = new PopupWithForm(
+    "#edit-avatar-popup",
+    handleAvatarFormSubmit
+  );
+
+  avatarImagePopup.setEventListeners();
+  document.querySelector(".profile__avatar").addEventListener("click", () => {
+    console.log("Avatar image clicked");
+    avatarImagePopup.open();
+  });
+
+  function handleImageClick(name, link) {
+    cardPreview.open({ name, link });
+  }
+
+  function createCard(cardData) {
+    const cardElement = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    );
+    return cardElement.getView();
+  }
+
+  const profileEditForm = new PopupWithForm("#profile-edit-popup", (data) => {
+    const { header, description } = data;
+    profileEditForm.renderLoading(true);
 
-cardSection.renderItems(constants.initialCards);
+    api
+      .updateUserInfo(header, description)
+      .then((res) => {
+        userInformation.setUserInfo(res);
+        userInformation.setUserAvatar(res);
+        profileEditForm.close();
+        profileEditValidator.disableButton();
+      })
+      .catch((err) => {
+        console.error("Error updating profile:", err);
+      })
+      .finally(() => {
+        profileEditForm.renderLoading(false);
+        profileEditValidator._enableButton();
```

Comment:
`        profileEditValidator._enableButton();
`

`_enableButton` is not needed in `finally` because the profile submit button can be disabled (no need to click it if you haven't changed the profile data in the inputs)

Reply:
`_enableButton`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.